aks-node-controller/app.go

```diff
@@ -95,6 +102,48 @@ func (a *App) Provision(ctx context.Context, flags ProvisionFlags) error {
 	return err
 }
 
+func (a *App) ProvisionWait(ctx context.Context) (string, error) {
+	if _, err := os.Stat(provisionJSONFilePath); err == nil {
+		data, err := os.ReadFile(provisionJSONFilePath)
+		if err != nil {
+			return "", err
+		}
+		return string(data), nil
+	}
+
+	watcher, err := fsnotify.NewWatcher()
+	if err != nil {
+		return "", fmt.Errorf("failed to create watcher: %w", err)
+	}
+	defer watcher.Close()
+
+	// Watch the directory containing the provision complete file
+	dir := filepath.Dir(provisionCompleteFilePath)
+	err = os.MkdirAll(dir, 0755) // create the directory if it doesn't exist
+	if err != nil {
+		return "", err
+	}
+	if err = watcher.Add(dir); err != nil {
+		return "", fmt.Errorf("failed to watch directory: %w", err)
+	}
+
+	for {
+		select {
+		case event := <-watcher.Events:
+			if event.Op&fsnotify.Create == fsnotify.Create && event.Name == provisionCompleteFilePath {
+				data, err := os.ReadFile(provisionJSONFilePath)
+				if err != nil {
+					return "", err
+				}
+				return string(data), nil
+			}
+
+		case err := <-watcher.Errors:
+			return "", fmt.Errorf("error watching file: %w", err)
+		}
```

Comment:
do we need a timeout?

Reply:
may be no necessary if ARM already have one on CSE execution

Reply:
actually, cancellation should be controlled through context.
I think we should just add something like:

```
case err := ctx.Done():
  return ctx.Err()
```